Source/WebKit/Shared/Extensions/_WKWebExtensionRegisteredScriptsSQLiteStore.mm

```diff
@@ -0,0 +1,295 @@
+            completionHandler(errorMessage);
+        });
+    });
+}
+
+- (void)getScriptsWithCompletionHandler:(void (^)(NSArray *, NSString *))completionHandler
+{
+    auto weakSelf = WeakObjCPtr<_WKWebExtensionRegisteredScriptsSQLiteStore> { self };
+    dispatch_async(_databaseQueue, ^{
+        NSString *errorMessage;
+        auto strongSelf = weakSelf.get();
+        auto* scripts = [strongSelf _getScriptsWithOutErrorMessage:&errorMessage];
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completionHandler(scripts, errorMessage);
+        });
+    });
+}
+
+- (NSArray *)_getScriptsWithOutErrorMessage:(NSString **)outErrorMessage
+{
+    dispatch_assert_queue(_databaseQueue);
+
+    if (![self _openDatabaseIfNecessaryReturningErrorMessage:outErrorMessage])
+        return @[ ];
+
+    ASSERT(!(*outErrorMessage).length);
+    ASSERT(_database);
+
+    _WKWebExtensionSQLiteRowEnumerator *rows = SQLiteDatabaseFetch(_database, @"SELECT * FROM registered_scripts");
+    return [self _getKeysAndValuesFromRowEnumerator:rows];
+}
+
+- (NSArray *)_getKeysAndValuesFromRowEnumerator:(_WKWebExtensionSQLiteRowEnumerator *)rows
+{
+    static NSSet *allowedClasses = [NSSet setWithObjects:NSString.class, NSArray.class, NSMutableDictionary.class, nil];
+
+    NSMutableArray<NSDictionary<NSString *, id> *> *scripts = [NSMutableArray array];
+
+    for (_WKWebExtensionSQLiteRow *row in rows) {
+        NSError *error;
+        NSDictionary<NSString *, id> *script = [NSKeyedUnarchiver unarchivedObjectOfClasses:allowedClasses fromData:[row dataAtIndex:1] error:&error];
+
+        if (error)
+            RELEASE_LOG_ERROR(Extensions, "Failed to deserialize registered content script for extension %{private}@.", _uniqueIdentifier);
+        else
+            [scripts addObject:script];
+    }
+
+    return scripts;
+}
+
+- (void)_insertScript:(NSDictionary *)script inDatabase:(_WKWebExtensionSQLiteDatabase *)database errorMessage:(NSString **)errorMessage
+{
+    dispatch_assert_queue(_databaseQueue);
+    ASSERT(_database);
+
+    NSData *scriptAsData = [NSKeyedArchiver archivedDataWithRootObject:script requiringSecureCoding:YES error:nil];
+    NSString *scriptID = script[idKey];
+    ASSERT(scriptID);
+
+    auto result = SQLiteDatabaseExecute(database, @"INSERT INTO registered_scripts (key, script) VALUES (?, ?)", scriptID, scriptAsData);
+    if (result != SQLITE_DONE) {
+        RELEASE_LOG_ERROR(Extensions, "Failed to insert registered content script for extension %{private}@.", _uniqueIdentifier);
+        *errorMessage =  @"Failed to add content script.";
+        return;
+    }
+}
+
+// MARK: Database Schema
+
+- (SchemaVersion)_currentDatabaseSchemaVersion
+{
+    return currentDatabaseSchemaVersion;
+}
+
+- (NSURL *)_databaseURL
+{
+    if (_useInMemoryDatabase)
+        return [_WKWebExtensionSQLiteDatabase inMemoryDatabaseURL];
+
+    ASSERT(_directory);
+
+    NSString *databaseName = @"registered-content-scripts.db";
```

Comment:
Can you rename this to "RegisteredContentScripts.db"? I'm working to clean up the files names we use.